plugins/src/ios/OptimoveSDKPlugin.swift

```diff
@@ -139,4 +193,266 @@ import NotificationCenter
         let pluginResult = CDVPluginResult(status: .ok)
         self.commandDelegate.send(pluginResult, callbackId: command.callbackId)
     }
+
+    // ========================== MESSAGING ==========================
+
+    @objc(setHandlersCallBackContext:)
+    func setHandlersCallBackContext(command: CDVInvokedUrlCommand){
+        OptimoveSDKPlugin.cordovaCommand = command
+
+        let checkForPendingPush: Bool = command.arguments[0] as! Bool
+        let checkForPendingDdl: Bool = command.arguments[1] as! Bool
+
+        if (checkForPendingPush){
+            self.checkIfPendingPushExists(command: command)
+        }
+
+        if (checkForPendingDdl){
+            //TODO
+
+            //return
+        }
+
+        let pluginResult:CDVPluginResult = CDVPluginResult.init(status: CDVCommandStatus_OK)
+        pluginResult.setKeepCallbackAs(true)
+        self.commandDelegate.send(pluginResult, callbackId: command.callbackId)
+    }
+
+    @objc(clearContext:)
+    func clearContext(command: CDVInvokedUrlCommand){
```

Comment:
When is this used?

Reply:
it's about cordova lifecycle, this one https://cordova.apache.org/docs/en/3.3.0/cordova/events/events.pause.html

I think on android this was necessary because when bg app native->js communication stops and we should be saving pending things